google/cloud/storage/object_stream.h

```diff
@@ -40,10 +40,14 @@ class ObjectReadStream : public std::basic_istream<char> {
    * Creates a stream associated with the given `streambuf`.
    */
   explicit ObjectReadStream(std::unique_ptr<internal::ObjectReadStreambuf> buf)
-      : std::basic_istream<char>(buf.get()), buf_(std::move(buf)) {}
+      : std::basic_istream<char>(buf.get()), buf_(std::move(buf)) {
+    peek();
```

Comment:
Could you add a comment saying why this peek is needed? This makes sense in the context of this PR, but I might not know why if I'm looking at this a year later :)